README.md

```diff
@@ -435,6 +355,347 @@ One App can be started via docker or if built from source by running `node lib/s
 
 <!-- TODO talk about prometheus and logging schema here -->
 
+## 🎛️ API
+
+### Overview
+
+#### [Modules](#modules)
+
+**Holoron Modules** or "Modules" are self contained Web Experiences that consist of React Components with Redux-compatible Reducers and Actions. In practice, Modules are developed and bundled in isolation to one another. The One App Server uses a Module Map containing URLs to Module bundles (e.g. `my-module.browser.js`) to load and serve bundles upon request. When the Server receives an HTTP request, it renders one or more Modules on the Server. Similar to React Components, Modules are composable (e.g. Modules may load other Modules). The first or entrypoint Module is called the "Root Module". The Root Module loads other "Child Modules". Overall, this development pattern in One App may be characterized as the [Micro Front End](https://martinfowler.com/articles/micro-frontends.html) pattern.
+
+**API**
+* [loadModuleData](#loadmoduledata)
+* [childRoutes](#childroutes)
+* [appConfig](#appconfig)
+
+#### [Server](#-server)
+
+TBD
+
+**API**
+* [TBD](#)
+
+---
+
+### Modules
+
+#### `loadModuleData`
+
+TBD
+
+#### `childRoutes`
+
+TBD
+
+#### `appConfig`
+
+`appConfig` allows a module to specify a selection of configuration options for Modules.
+
+```js
+// Force tree shaking appConfig away in client bundles
+if (!global.BROWSER) {
+  MyModule.appConfig = {
+    /* Tenant Root Specific */
+    provideStateConfig,
+    csp,
+    corsOrigins,
+    configureRequestLog,
+    extendSafeRequestRestrictedAttributes,
+    createSsrFetch,
+    /* Child Module Specific */
+    validateStateConfig,
+    requiredSafeRequestRestrictedAttributes,
+    /* All Modules */
+    appCompatibility,
+  };
+}
+```
+
+**Contents**
+* [provideStateConfig](#providestateconfig)
+* [csp](#csp)
+* [corsOrigins](#corsorigins)
+* [configureRequestLog](#configurerequestlog)
+* [extendSafeRequestRestrictedAttributes](#extendsaferequestrestrictedattributes)
+* [createSsrFetch](#createssrfetch)
+* [validateStateConfig](#validatestateconfig)
+* [requiredSafeRequestRestrictedAttributes](#requiredsaferequestrestrictedattributes)
+* [appCompatibility](#appcompatibility)
+
+##### `provideStateConfig`
+**Module Type**
+* ✅ Root Module 
+* 🚫 Child Module  
+
+**Shape**
+```js
+Module.appConfig = {
```

Comment:
is this shape too late to adjust? `appConfig` `stateConfig` config, config, config 

```
appConfig: {
  abc,
  def,
}
```